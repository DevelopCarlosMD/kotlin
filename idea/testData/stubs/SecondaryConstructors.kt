package test

class SecondaryConstructors(x: Boolean) {
    init {
    }

    constructor(x: String) : this(x == "abc") {
    }

    init {
    }

    constructor(x: Int) : this(x < 0) {
    }
}
