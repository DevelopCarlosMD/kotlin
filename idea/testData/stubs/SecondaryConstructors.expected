PsiJetFileStubImpl[package=test]
  PACKAGE_DIRECTIVE:
    REFERENCE_EXPRESSION:[referencedName=test]
  CLASS:[fqName=test.SecondaryConstructors, isEnumEntry=false, isLocal=false, isTopLevel=true, isTrait=false, name=SecondaryConstructors, superNames=[]]
    VALUE_PARAMETER_LIST:
      VALUE_PARAMETER:[fqName=null, hasDefaultValue=false, hasValOrVarNode=false, isMutable=false, name=x]
        TYPE_REFERENCE:
          USER_TYPE:[isAbsoluteInRootPackage=false]
            REFERENCE_EXPRESSION:[referencedName=Boolean]
    CLASS_BODY:
      ANONYMOUS_INITIALIZER:
      SECONDARY_CONSTRUCTOR:
        VALUE_PARAMETER_LIST:
          VALUE_PARAMETER:[fqName=null, hasDefaultValue=false, hasValOrVarNode=false, isMutable=false, name=x]
            TYPE_REFERENCE:
              USER_TYPE:[isAbsoluteInRootPackage=false]
                REFERENCE_EXPRESSION:[referencedName=String]
        CONSTRUCTOR_DELEGATION_CALL:
      ANONYMOUS_INITIALIZER:
      SECONDARY_CONSTRUCTOR:
        VALUE_PARAMETER_LIST:
          VALUE_PARAMETER:[fqName=null, hasDefaultValue=false, hasValOrVarNode=false, isMutable=false, name=x]
            TYPE_REFERENCE:
              USER_TYPE:[isAbsoluteInRootPackage=false]
                REFERENCE_EXPRESSION:[referencedName=Int]
        CONSTRUCTOR_DELEGATION_CALL:
